BuildAddOn.py

```diff
@@ -277,8 +277,18 @@ def CopyResultToPackage (packageRootFolder, buildFolder, version, addOnName, pla
         ])
 
 
+# Extract the DevKit build numbers for each ACVersion
+def GetBuildNums (acVersionList, devKitFolderList):

```

Comment:
This function is very error prone. Any changes in the package folder name can break this. Any other zip file in the directory will break this. This is not something I would happily to release to every third-party developers. Why do we need the build number in the zip file name?

Reply:
The core problem: Add-On builds are generated during the daily build, these Add-On will be used in in-house testing - currently a tester that gets an Add-On build cannot decide which Archicad is it compatible with.
Automatic Add-On integration is not our scope anymore, but testers should be able to decide what Archicad is an Add-On build compatible with - relatively painlessly, if possible. 
The Add-Ons are built with DevMain DevKits, this is a fixed parameter - the build num of the AC build from which the DevKit comes from will clearly define the compatibility.
The buildnum should be included somewhere into the Add-On build - deciding compatibility is possible.
Including the DevKit buildnum into the name - deciding compatibility is easy.
This given solution is fragile, I agree, we need to make it a robust solution. Passing the needed version and buildnum version in the APIDevKit would make it more robust, this solution is already present in the win APIDevKit package, but not in the mac one. Currently the name is the only way we can get the APIdevKit build num from a Mac APIDevKit. And as stated above we will need to know the buildnum to decide compatibility.
I will look into what versioning option there are and which one would be a satisfying solution to our problem.

Reply:
I agree that this is something we need to solve. The complexity here comes from the fact that users can use any url for the DevKit, and not only the ones we provide. To cut this short, my proposal is to make this work only with the officially provided DevKits where we are in control of the naming. This is a limitation for third-parties, but we don't really know any workflows where an unofficial location would be useful.

Reply:
We can rename the created package in the daily pipeline after packages are put together and that way only the officially provided DevKits would be used and only in case of our repos and pipelines. This solves the problem of giving it to third-party developers and we are in control of what APIDevKits are used. 
At the same time the correct solution would be finding a good way to pass the buildnum forward from Archicad to APIDevKit package to Add-On. I would like to bring this up in a APICouncil meeting.